annotation/consume_trigger.go

```diff
@@ -1562,13 +1836,18 @@ type UseAsErrorRetWithNilabilityUnknownPrestring struct {
 	RetNum        int
 	IsNamedReturn bool
 	RetName       string
+	AssignmentStr string
 }
 
 func (u UseAsErrorRetWithNilabilityUnknownPrestring) String() string {
+	var sb strings.Builder
 	if u.IsNamedReturn {
-		return fmt.Sprintf("found in at least one path of `%s()` for named return `%s` in position %d", u.FuncName, u.RetName, u.RetNum)
+		sb.WriteString(fmt.Sprintf("found in at least one path of `%s()` for named return `%s` in position %d", u.FuncName, u.RetName, u.RetNum))
+	} else {
+		sb.WriteString(fmt.Sprintf("found in at least one path of `%s()` for return in position %d", u.FuncName, u.RetNum))
 	}
-	return fmt.Sprintf("found in at least one path of `%s()` for return in position %d", u.FuncName, u.RetNum)
+	sb.WriteString(u.AssignmentStr)
+	return sb.String()
```

Comment:
Having read through this all, I'm actually a bit tempted to create a shared struct, say `metaPrestring` or something similar that stores `AssignmentStr` since it is shared by all of the prestring nodes. Then we just embed this struct in all Prestring structs.

Then, this `metaPrestring` can expose two methods, `writePrefix(io.Writer)` and `writeSuffix(io.Writer)`, which is called at the beginning and at the end of each `String()` function (by `u.meta.writePrefix(sb)`, since `strings.Builder` implements the `io.Writer` interface).

This would allow future expansion a little easier. 

But I also see that might be an overkill for now, so we can keep this in mind until we need to add another field to all prestrings :)

Reply:
Totally agreed! Every time I visit `consume_trigger.go` and `produce_trigger.go`, I am tempted to rearchitect it for Prestrings. I feel there is a lot of repetition there. A problem to be solved some other day :)